cpp/src/rolling/rolling_detail.cuh

```diff
@@ -1062,6 +1062,72 @@ struct rolling_window_launcher {
     return gather_map;
   }
 
+  /**
+   * @brief Purge entries for null inputs from gather_map, and adjust offsets.
+   */
+  void purge_null_entries(column_view const& input,
+                          std::unique_ptr<column>& gather_map,
+                          std::unique_ptr<column>& offsets,
```

Comment:
Passing a `unique_ptr` by non-const reference declares that you intend to reseat that pointer. That doesn't appear to be the case here.

https://www.modernescpp.com/index.php/c-core-guidelines-passing-smart-pointer

Reply:
The `gather_map` and `offsets` are reseated at lines 1128-1129:
```c++
    ...
    gather_map = std::move(new_gather_map);
    offsets    = std::move(new_offsets);
```

Reply:
Ah I see. Any reason the old gather map/offsets can't be an input and the new ones be an output? That would be more canonical with libcudf style.

Reply:
> the new ones be an output... more canonical with libcudf style.

I shied from returning `pair<unique_ptr<column>, unique_ptr<column>>`. Let me change this around.

Reply:
Fixed, at last. Some of the surrounding code needed moving around to accommodate the change. The null counting needed to be moved out of the function, etc.